eth_typing/abi.py

```diff
@@ -216,8 +224,20 @@ class ABIFunctionInfo(TypedDict, total=False):
     """Function input components."""
 
 
-ABIElement = Union[ABICallable, ABIEvent, ABIError]
-"""Base type for `ABIFunction` and `ABIEvent` types."""
+class ABIElementInfo(TypedDict):
```

Comment:
All attributes are required here, right? Or put another way,`total=False` does not apply?

Reply:
Thanks for checking, yes the attributes should always be present. Just for context, this type was created to fix the Tuple that was returned by (`get_abi_element_info`)[https://github.com/ethereum/web3.py/blob/bb2ef6e122faab60251222be983905010a3a2cc4/web3/_utils/contracts.py#L361].

Here's where I'm using it: https://github.com/ethereum/web3.py/blob/f00e889adb830726c793c65c096a64e07f323179/web3/utils/abi.py#L184

Reply:
🚀